sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -140,28 +140,33 @@ private[sql] trait LookupCatalog extends Logging {
   }
 
   /**
-   * Extract catalog and the rest name parts from a multi-part identifier.
+   * Extract catalog and the rest name parts as a table identifier.
+   * This does not return an empty array as a table name, thus for the single-part name,
+   * it is resolved to a table name, not a catalog.
    */
-  object CatalogAndIdentifierParts {
+  object CatalogAndTable {
     private val globalTempDB = SQLConf.get.getConf(StaticSQLConf.GLOBAL_TEMP_DATABASE)
 
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = {
```

Comment:
shall we return `Identifer` instead of `Seq[String]`? Then it's guaranteed that we won't return invalid table name.